api/python/t4/packages.py

```diff
@@ -618,30 +597,29 @@ def set(self, logical_key, entry=None, meta=None):
         Returns:
             self
         """
-        if entry is None and meta is None:
-            raise PackageException('Must specify either entry or meta')
-
-        if entry is None:
-            return self._update_meta(logical_key, meta)
-
-        if isinstance(entry, (string_types, binary_type, getattr(os, 'PathLike'))):
+        if isinstance(entry, (string_types, binary_type, getattr(os, 'PathLike', str))):
             entry = PackageEntry.from_local_path(entry)
-            if meta is not None:
-                entry.meta = meta
-            self._data[logical_key] = entry
         elif isinstance(entry, PackageEntry):
-            if meta is not None:
-                raise PackageException("Must specify metadata in the entry")
-            self._data[logical_key] = entry
+            entry = entry._clone()
         else:
             raise NotImplementedError("Needs to be of type str")
```

Comment:
Missing a word for clarity `"Expected a string for X"` or `"X needs to be  of type str"`